### PR TITLE
[MC Garden] - Feature/mc garden setup

### PR DESCRIPTION
**Pull request by jaysencpp**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: [AB#203801](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/203801)
Completes: [AB#203839](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/203839)

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/MechanicalCompletion/components/CustomHeaderView/CustomHeaderView.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { CustomHeaderView } from '@equinor/ParkView';
+import { memo } from 'react';
+import { McPackage } from '../../types';
+import { HeaderContainer } from './CustomHeaderView.styles';
+
+const McGardenHeader = ({ garden, columnIndex, groupByKey }: CustomHeaderView<McPackage>) => {
+    const label =
+        groupByKey === 'finalPunch'
```
Comment (Gustav-Eikaas) [inline src/apps/MechanicalCompletion/components/CustomHeaderView/CustomHeaderView.tsx:8]: Consider switch statement

File: src/apps/MechanicalCompletion/utils/helpers/getStatuses.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    } else if (mcPackage.rfccIsRejected) {
+        return 'RFCC Rejected';
+    } else if (mcPackage.rfccIsShipped) {
+        return 'RFCC Shipped';
+    } else if (mcPackage.punchAcceptActualDate) {
+        return 'Punch status accepted';
+    } else if (mcPackage.finalPunchActualDate) {
+        return 'Contractor final punch';
+    }
+    return 'OS';
```
Comment (Gustav-Eikaas) [inline src/apps/MechanicalCompletion/utils/helpers/getStatuses.ts:25]: Consider switch statement

File: src/packages/GardenUtils/src/hooks/usePackageResource.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    useEffect(() => {
+        preformFetch();
+    }, [preformFetch]);
+
+    return {
+        data,
+        isFetching,
+        error,
+        reset,
+    };
```
Comment (Gustav-Eikaas) [inline src/packages/GardenUtils/src/hooks/usePackageResource.ts:42]: Consider using react-query

**Review by Gustav-Eikaas — APPROVED** [review]
LGTM

File: src/packages/GardenUtils/src/hooks/usePackageResource.ts (hunk 3 above)
Comment (jaysencpp) [reply src/packages/GardenUtils/src/hooks/usePackageResource.ts:42]: will do with all the garden queries in another PR

**Pull request merged**